### PR TITLE
Mohamed chat message CRUD

### LINKED ISSUE #32: CRUD for Chat Messages
#5 Implement a Controller for chat messages. get and post

### PR DESCRIPTION
**Pull request by Mohamed-Elfouly**
closes #32 

Implements the controller for chat messages

Mutation testing
image

### CODE REVIEW

**Comment by cjyu81** [comment]
good to merge

**Review by cjyu81 — APPROVED** [review]
Looks good. Approving

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+public class ChatMessageController extends ApiController {
+
+    @Autowired
+    ChatMessageRepository chatMessageRepository;
+
+    @Operation(summary = "Create a new message")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @PostMapping("/post")
+    public ChatMessage postMessage(
+        @Parameter(name="content", description="String, Time the ride starts HH:MM(A/P)M", example="12:30AM", required = true)
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java:47]: I wonder if this description and example are accurate?  It seems to have been copied/pasted from another description.

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        @RequestParam String content
+        )
+        {
+
+        ChatMessage message = new ChatMessage();
+        
+        message.setUserId(getCurrentUser().getUser().getId());
+        message.setPayload(content);
+        //message.setDm(false);
+        //message.setToUserId(null);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java:57]: we should uncomment these;  This is one place where a comment saying that the fields are "for future use" might be warranted.
```suggestion
        message.setDm(false); // future feature; not currently used
        message.setToUserId(null);  // future feature; not currently used
```

**Review by pconrad — CHANGES_REQUESTED** [review]
A couple of things to look at

**Comment by Mohamed-Elfouly** [comment]
> A couple of things to look at

Now mutation testing fails. Am I still meant to test features that are to be made later?

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java (hunk 1 above)
Comment (Mohamed-Elfouly) [reply src/main/java/edu/ucsb/cs156/gauchoride/controllers/ChatMessageController.java]: I changed the description

**Review by pconrad — APPROVED** [review]

**Pull request merged**